### DISCUSSION TITLE
Where is the place that mxnet call cblas_gemm if I use openblas?

### DISCUSSION THREAD

**Discussion by knjwhn** [apache/incubator-mxnet#16557] [Q&A] [answered]
Hello,guys,I made some modification in openblas (add a new function cblas_mygemm)and I wanna to use it in mxnet , is there have a place or something to call cblas_sgemm() in mxnet source code ? Hope for your help.

**Comment by roywei** [comment]
cc @larroy

**Comment by larroy** [comment]
Hi @knjwhn 

The blas calls are generated with some preprocessor macro magic as you can see from here:

https://github.com/apache/incubator-mxnet/blob/master/src/operator/linalg_impl.h#L149

I think you might be able to do what you want around these functions.

**Comment by knjwhn** [comment]
Thanks! and there still a question, what if my gemmlib  only contains cblas_sgemm  , can it pass the compile? because i see there is another function like trmm.

**Comment by larroy** [comment]
could you clarify your question? You mean there's two functions with the same name when linking?

**Comment by knjwhn** [comment]
yeah, I made a new sgemm function(name not same as cblas_sgemm) in a lite version Openblas(I called gemmlib),and I want use it in mxnet,  Do you have some suggestion？Many thanks！

**Comment by larroy** [comment]
Hi @knjwhn I think you want to link mxnet with your library and call the function in the dispatcher for the linalg operations if I understood your intention correctly.

**Comment by knjwhn** [comment]
Yeah,that’s my intention, I want to test it in DNN training (especially in convolution),Do you have some suggestion that how can I change the makefile?



发自我的iPhone


------------------ Original ------------------

**Comment by larroy** [comment]
You can see how libraries are added in the CMakefile, for make just link with your library. Let me know if you have issues. That should be straighforward.

**Comment by larroy** [comment]
in cmake you can use `add_library`

**Comment by knjwhn** [comment]
Thanks a lot , I modified the makefile and compile it successsfully. and I run the example/image-classfication/train_mnist.py ,However  I don't know if my sgemm is called? What should I do?  and There is another question , I wrote an int8 gemm function(A:u8/s8 B:u8/s8 C:int32),can it use in convolution calculation after quantization? Hope for your help.

**Comment by wkcn** [comment, marked as answer]
Hi @knjwhn,

https://github.com/apache/incubator-mxnet/blob/60d74bc948869588c2f143fd3d55231859dc979f/src/operator/linalg_impl.h#L149

**Comment by knjwhn** [comment]
> Hi @knjwhn,
> 
> https://github.com/apache/incubator-mxnet/blob/60d74bc948869588c2f143fd3d55231859dc979f/src/operator/linalg_impl.h#L149

Thanks,I've done that.  and instead of float32 type . I wrote an int8 openblas gemm function(A:u8/s8 B:u8/s8 C:int32),can it use in convolution calculation after quantization? Hope for your help.
